crates/store/src/db/models/queries/accounts.rs

```diff
@@ -541,6 +485,219 @@ pub(crate) fn select_account_storage_map_values(
     Ok(StorageMapValuesPage { last_block_included, values })
 }
 
+/// Reconstruct a `StorageMap` from database entries using `SmtForest`
+///
+/// This function builds an `SmtForest` from all key-value pairs at the specified block,
+/// enabling efficient proof generation with structural sharing. The forest allows
+/// maintaining multiple SMT versions in memory with shared nodes.
```

Comment:
I don't think we are using the `SmtForest` here (which I think is fine for now).

For the future, `SmtForest` would live in a different place - it would be at the same level where we currently have account and nullifier trees. And once we have it, we should be able to get rid of these queries.